FinSearchUnified/Bundle/SearchBundleFindologic/QueryBuilder.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace FinSearchUnified\Bundle\SearchBundleFindologic;
+
+use Exception;
+use Shopware\Bundle\PluginInstallerBundle\Service\InstallerService;
+use Shopware\Components\HttpClient\HttpClientInterface;
+use Shopware\Components\HttpClient\RequestException;
+use Shopware\Components\HttpClient\Response;
+use Shopware_Components_Config;
+
+class QueryBuilder
+{
+    const BASE_URL = 'https://service.findologic.com/ps/xml_2.0/';
+    const ALIVE_ENDPOINT = 'alivetest.php';
+    const SEARCH_ENDPOINT = 'index.php';
+    const NAVIGATION_ENDPOINT = 'selector.php';
+
+    /**
+     * @var HttpClientInterface
+     */
+    protected $httpClient;
+
+    private $parameters;
+
+    private $shopUrl;
+
+    private $shopKey;
+
+    /**
+     * QueryBuilder constructor.
+     *
+     * @param HttpClientInterface $httpClient
+     * @param InstallerService $installerService
+     * @param Shopware_Components_Config $config
+     *
+     * @throws Exception
+     */
+    public function __construct(
+        HttpClientInterface $httpClient,
+        InstallerService $installerService,
+        Shopware_Components_Config $config
+    ) {
+        $plugin = $installerService->getPluginByName('FinSearchUnified');
+
+        $this->parameters = [];
+        $this->httpClient = $httpClient;
+        $this->shopKey = $config->offsetGet('ShopKey');
+        $this->shopUrl = rtrim(Shopware()->Shop()->getHost(), '/') . '/';
+        $this->parameters = [
+            'userip' => $this->getClientIp(),
+            'revision' => $plugin->getVersion(),
+            'shopkey' => $this->shopKey
+        ];
+    }
+
+    /**
+     * @param bool $isSearch
+     *
+     * @return string|null
+     */
+    public function execute($isSearch = true)
+    {
+        if ($isSearch) {
+            $endpoint = self::SEARCH_ENDPOINT;
+        } else {
+            $endpoint = self::NAVIGATION_ENDPOINT;
+        }
+
+        $url = sprintf(
+            '%s%s%s?%s',
+            self::BASE_URL,
+            $this->shopUrl,
+            $endpoint,
+            http_build_query($this->parameters)
+        );
+
+        try {
+            if ($this->isAlive()) {
+                $response = $this->httpClient->get($url);
+
+                return $response->getBody();
```

Comment:
I think we can refactor this a little bit:
```
$payload = null;

try {
    if ($this->isAlive()) {
        $response = $this->httpClient->get($url);
        $payload = $response->getBody();
    }
} catch (RequestException $exception) {
}

return $response;
```

Usually we prefer to have a `else` when there is a `if` but in this case, it makes the code more readable when we remove some stuff. Unless you have a different opinion?